### PR TITLE
Add acceptance test steps related to trusted server

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add Test steps for handling trusted servers.

## Related Issue

- Fixes <issue_link>

## Motivation and Context

API for handling trusted servers was added to the testing app in https://github.com/owncloud/testing/pull/101 along with its test steps. Since there is no other way to add/remote trusted servers other than API these steps are good to have in core.

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/AppConfiguration.php [hunk 1]
```diff
@@ -378,6 +386,183 @@ protected function setStatusTestingApp($enabled) {
...
+	 *
+	 * @param string $url
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasAddedUrlAsTrustedServer($url) {
+		$this->theAdministratorAddsUrlAsTrustedServerUsingTheTestingApi($url);
+		$status = $this->getResponse()->getStatusCode();
+		if ($status !== 201) {
+			throw new \Exception("The request failed with status $status");
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/AppConfiguration.php]: maybe say what request failed

File: tests/acceptance/features/bootstrap/AppConfiguration.php [hunk 2]
```diff
@@ -378,6 +386,183 @@ protected function setStatusTestingApp($enabled) {
...
+	 *
+	 * @param string $url
+	 *
+	 * @return void
+	 */
+	public function urlShouldNotBeATrustedServer($url) {
+		$trustedServers = $this->getTrustedServers();
+		foreach ($trustedServers as $server => $id) {
+			if ($server === $url) {
+				\PHPUnit\Framework\Assert::fail("Given url $url is a trusted server");
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/AppConfiguration.php]:
```suggestion
				\PHPUnit\Framework\Assert::fail("Given url $url is a trusted server, but should not be.");
```

File: tests/acceptance/features/bootstrap/AppConfiguration.php [hunk 3]
```diff
@@ -408,6 +593,24 @@ public function prepareParametersBeforeScenario() {
...
+	}
+
+	/**
+	 * After Scenario. restore trusted servers
+	 *
+	 * @AfterScenario
+	 *
+	 * @return void
+	 */
+	public function restoreTrustedServers() {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/AppConfiguration.php]: we will have to run that on both remote an local server

**Review by individual-it — CHANGES_REQUESTED** [review]
a couple of small comments
where are the tests themself? different PR?

**Comment by dpakach** [comment]
> a couple of small comments
> where are the tests themself? different PR?

The tests will be added in different PR. Tests using these steps are in testing app and are passing locally.

**Review by individual-it — APPROVED** [review]

**Comment by phil-davis** [comment]
CI failed https://drone.owncloud.com/owncloud/core/19395/618
```
  Scenario: Delete a user, and specify the user name in different case         # /drone/src/tests/acceptance/features/cliProvisioning/deleteUser.feature:16
    Given these users have been created with skeleton files:                   # FeatureContext::theseUsersHaveBeenCreated()
      | username       |
      | brand-new-user |
    When the administrator deletes user "Brand-New-User" using the occ command # OccUsersGroupsContext::theAdministratorDeletesUserUsingTheOccCommand()
    Then the OCS status code should be "100"                                   # OCSContext::theOCSStatusCodeShouldBe()
      OCS status code is not the expected value
      Failed asserting that two strings are equal.
      --- Expected
      +++ Actual
      @@ @@
      -'100'
      +'200'
    And the HTTP status code should be "200"                                   # FeatureContext::theHTTPStatusCodeShouldBe()
    And user "brand-new-user" should not exist                                 # FeatureContext::userShouldNotExist()
```
How did the code in this PR effect that?

Restarting drone CI...

**Review by phil-davis — APPROVED** [review]

**Comment by dpakach** [comment]
Backport on #35823

**Pull request merged**